0009-comit-btsieve-db.adoc

```diff
@@ -0,0 +1,253 @@
+= Persistent database for comit_node and btsieve
+Thomas Eizinger <thomas@coblox.tech>;
+:toc:
+:revdate: 2019-07-12
+
+NOTE: Author: {authors} +
+Date: {revdate} +
+Tracking issue: https://github.com/comit-network/comit-rs/issues/1060[#1060]
+
+== Context
+
+Currently, `comit_node` as-well as `btsieve` store all of their data just in-memory, which means it is lost as soon as the process stops.
+This is obviously not acceptable for software that deals with money and where the state being stored (transient keypairs, pre-images, etc) is crucial in accessing it.
+
+== Goals
+
+1. No separate process
+
+One of the core development goals of COMIT is to be as easily accessible as possible.
+Hence, any solution that involves running a _separate_ process for persistently storing data is not considered.
+All the solutions proposed here need to be embeddable into `comit_node` and `btsieve`.
+
+2. Atomic access
+
+Data integrity is key to our software.
+Hence, all access to the data needs be done in an atomic fashion.
+
+== What do we want to store?
+
+In order to decide on the best library/solution for storing data in `comit_node` and `btsieve` we need to know, what data we want to store and how we would like to store/access it.
+
+=== At the time of writing, we want to store the following things:
+
+==== `comit_node`
+
+- A list of queries created on `btsieve`
+- A list of swaps +
+This sounds less trivial than it is.
```

Comment:
Did you mean 'This is more difficult than it sounds'?

Reply:
Yes.